make/modules/java.base/Lib.gmk

```diff
@@ -203,3 +203,40 @@ ifeq ($(call isTargetOs, windows), true)
   ))
   TARGETS += $(COPY_TZMAPPINGS)
 endif
+
+################################################################################
+# Create system lookup
+
+ifeq ($(call isTargetOs, linux), true)
+
+  $(eval $(call SetupJdkLibrary, BUILD_SYSLOOKUPLIB, \
+      NAME := syslookup, \
+      CFLAGS := $(CFLAGS_JDKLIB), \
+      CXXFLAGS := $(CXXFLAGS_JDKLIB), \
+      LDFLAGS := $(LDFLAGS_JDKLIB) -Wl$(COMMA)--no-as-needed, \
+      LIBS := $(LIBCXX) -lc -lm -ldl, \
```

Comment:
Instead of repeating this whole macro call for both Linux and non Linux, you can use parameters of the form LDFLAGS_linux and LIBS_linux to add the Linux specific flags. Something like this:

```
LDFLAGS := $(LDFLAGS_JDKLIB), \
LDFLAGS_linux := -Wl$(COMMA)--no-as-needed, \
```

For the NAME field, there is no such support, so the way we usually do that is through a variable and conditionals before the macro call. What's the reason to have a different lib name on Windows? If they were the same, and the source file in windows/native/... had the same name, it would just automatically override in the build.

I realize now that this is just moved code from jdk.incubator.foreign, and this patch is probably big enough as it is so no need to refactor the build logic at the same time.

Reply:
Good points - there is really no need AFAIK for the lib name to be different. I'll do few experiments.

Reply:
I've fixed the makefile as you suggested - I agree the result is much simpler. I've tested the changes on mac/linux/win and everything looks good.